tests/std/tests/P1208R6_source_location/test.cpp

```diff
@@ -5,10 +5,17 @@
 #include <cassert>
 #include <functional>
 #include <source_location>
+#include <string>
 #include <string_view>
 #include <type_traits>
 using namespace std;
 
+#ifdef _M_IX86
+#define THISCALL_OR_CDECL_STR "__thiscall"s

```

Comment:
I note that all uses of `THISCALL_OR_CDECL_STR` concatenate it with string literals. We _could_ define this as a string literal and let the parser do the concatenation for us (replace `"meow" + THISCALL_OR_CDECL_STR + "woof"` with `"meow" THISCALL_OR_CDECL_STR "woof"`).

Absent that, there seems to be no reason to use a macro when 
```c++
constexpr string thiscall_or_cdecl{
#ifdef _M_IX86
    "__thiscall"
#else
    "__cdecl"
#endif
};
```
would work as well.

No change requested, because (1) we don't really have a convention, this is just style, and (2) I feel like I've already asked for enough work in this PR.

Reply:
As I mentioned on Discord:

* We can't just say `"meow" MACRO "woof"` because `function_name()` returns `const char*`, so we need a `string_view` or a `string` to compare against.
* We can't have a top-level `constexpr string` because `constexpr` dynamic allocations can't survive until runtime.

However, `"meow" MACRO "woof"sv` works. I went ahead and pushed a change to do that, because `constexpr string` is a very high-powered feature (read: lots of compiler bugs historically) and this test is already stressing the compiler enough.